test/jdk/javax/swing/JFileChooser/FileChooserSizeLabelCutOffTest.java

```diff
@@ -0,0 +1,139 @@
+import javax.swing.SwingUtilities;
+import javax.swing.UIManager;
+import javax.swing.WindowConstants;
+
+import java.util.function.Predicate;
+
+/*
+ * @test
+ * @bug 8301606
+ * @key headful
+ * @library /java/awt/regtesthelpers
+ * @build PassFailJFrame
+ * @summary Test to check if the Size label in details view,
+ * doesn't cut off Metal Look&Feel
+ * @run main/manual/othervm -Dsun.java2d.uiScale=2.25 FileChooserSizeLabelCutOffTest
+ */
+
+public class FileChooserSizeLabelCutOffTest {
+    static JFrame frame;
+    static JFileChooser jfc;
+    static PassFailJFrame passFailJFrame;
+
+    public static void main(String[] args) throws Exception {
+        UIManager.setLookAndFeel("javax.swing.plaf.metal.MetalLookAndFeel");
+        Robot robot = new Robot();
+        SwingUtilities.invokeAndWait(new Runnable() {
+            public void run() {
+                try {
+                    initialize();
+                } catch (Exception e) {
+                    throw new RuntimeException(e);
+                }
+            }
+        });
+        robot.delay(500);
+        robot.waitForIdle();
+        SwingUtilities.invokeAndWait(new Runnable() {
+            public void run() {
+                switchToDetailsView();
+            }
+        });
+        robot.delay(500);
+        robot.waitForIdle();
+        passFailJFrame.awaitAndCheck();
+    }
+
+    static void initialize() throws Exception {
+        final String INSTRUCTIONS = """
+                Instructions to Test:
+                1. Verify that the UI scaling is set to 225%.
+                2. If Size label is cut off with table
+                 header cell, test FAIL else test is PASS.
+                """;
+        frame = new JFrame("JFileChooser Size Label test");
+        jfc = new JFileChooser();
+        passFailJFrame = new PassFailJFrame("Test Instructions", INSTRUCTIONS,
+                5L, 8, 25);
+
+        PassFailJFrame.addTestWindow(frame);
+        PassFailJFrame.positionTestWindow(frame, PassFailJFrame.Position.TOP_LEFT_CORNER);
+
+        frame.setDefaultCloseOperation(WindowConstants.EXIT_ON_CLOSE);
+        jfc.setControlButtonsAreShown(false);
+        jfc.setDialogType(JFileChooser.CUSTOM_DIALOG);
+
+        frame.add(jfc, BorderLayout.CENTER);
+        frame.pack();
+        frame.setVisible(true);
+    }
+    private static void switchToDetailsView() {
+
+        AbstractButton detailsBtn = findDetailsButton(jfc);
+        if (detailsBtn == null) {
+            throw new Error("'Details' button not found in JFileChooser");
+        }
+        detailsBtn.doClick();
+    }
+    private static AbstractButton findDetailsButton(final Container container) {
+        Component result = findComponent(container,
+                c -> c instanceof JToggleButton button
+                        && "Details".equals(button.getToolTipText()));
+        return (AbstractButton) result;
+    }
+
+    private static JTable findTable(final Container container) {
```

Comment:
This method is not called at any place, can it be removed?

Reply:
Yeah, was actually trying to do that but couldn't get exact coordinates of header cell. Later on forgot to remove the unused methods.

Reply:
And also it's not possible to verify with background color because anti-aliasing, I guess.

Reply:
> And also it's not possible to verify with background color because anti-aliasing, I guess.

Why is it? There's no background anti-aliasing, the background is filled with solid color.

You need to use [Robot.createMultiResolutionScreenCapture](https://docs.oracle.com/en/java/javase/17/docs/api/java.desktop/java/awt/Robot.html#createMultiResolutionScreenCapture(java.awt.Rectangle)) to get the real screen pixels rather than scaled down image as if uiScale=1.0.

You may need to tweak the position of the capture and the area where you sample the pixels.

Reply:
Yeah, verifying with JTable will be better idea.